packages/drivers/routerlicious-driver/src/restWrapper.ts

```diff
@@ -152,13 +181,29 @@ export class RouterliciousRestWrapper extends RestWrapper {
 				// the error message will start with NetworkError as defined in restWrapper.ts
 				// If there exists a self-signed SSL certificates error, throw a NonRetryableError
 				// TODO: instead of relying on string matching, filter error based on the error code like we do for websocket connections
-				const err = errorMessage.includes("failed, reason: self signed certificate")
-					? new NonRetryableError(errorMessage, RouterliciousErrorTypes.sslCertError, {
-							driverVersion,
-						})
-					: new GenericNetworkError(errorMessage, errorMessage.startsWith("NetworkError"), {
-							driverVersion,
-						});
+				const err = errorMessage.includes(
+					"failed, reason: self signed certificate",
+				)
+					? new NonRetryableError(
+							errorMessage,
+							RouterliciousErrorTypes.sslCertError,
+							{
+								driverVersion,
+							},
+						)
+					: new GenericNetworkError(
+							errorMessage,
+							errorMessage.startsWith("NetworkError"),
+							{
+								driverVersion,
+							},
+						);
+
+				// on failure, add the request entry into the retryCounter map to count the subsequent retries
+				this.retryCounter.set(
```

Comment:
Do we intend to use this map elsewhere? Like in client side Telemetry?

Reply:
Not the map, but I have now logged the value of the retry query param sent to the server to our error logs atleast, so that we have some insights if needed.